src/region_map.c

```diff
@@ -2025,3 +2015,32 @@ static void CB_ExitFlyMap(void)
         break;
     }
 }
+
+u32 FilterFlyDestination(struct RegionMap* regionMap)
+{
+    switch (regionMap->mapSecId)
+    {
+    case MAPSEC_SOUTHERN_ISLAND:
+        return HEAL_LOCATION_SOUTHERN_ISLAND_EXTERIOR;
+    case MAPSEC_BATTLE_FRONTIER:
+        return HEAL_LOCATION_BATTLE_FRONTIER_OUTSIDE_EAST;
+    case MAPSEC_LITTLEROOT_TOWN:
+        return (gSaveBlock2Ptr->playerGender == MALE ? HEAL_LOCATION_LITTLEROOT_TOWN_BRENDANS_HOUSE : HEAL_LOCATION_LITTLEROOT_TOWN_MAYS_HOUSE);
+    case MAPSEC_EVER_GRANDE_CITY:
+        return (FlagGet(FLAG_LANDMARK_POKEMON_LEAGUE) && regionMap->posWithinMapSec == 0 ? HEAL_LOCATION_EVER_GRANDE_CITY_POKEMON_LEAGUE : HEAL_LOCATION_EVER_GRANDE_CITY);
+    default:
+        if (sMapHealLocations[regionMap->mapSecId][2] != HEAL_LOCATION_NONE)
+            return sMapHealLocations[regionMap->mapSecId][2];
+        else
+            return WARP_ID_NONE;
+    }
+}
+
+void SetFlyDestination(struct RegionMap* regionMap)
+{
+    u8 flyDestination = FilterFlyDestination(regionMap);
```

Comment:
```suggestion
    u32 flyDestination = FilterFlyDestination(regionMap);
```

Reply:
doesn't this one need to be a u8? `void SetWarpDestinationToHealLocation(u8 healLocationId);`

Reply:
No, it doesn't. The compiler waste an instruction an turns it into a u32 anyways so technically it would be better to change the healLocationId argument to a u32.

Reply:
ah rad- will adjust! ty!

Reply:
should `healLocationId`-> u32 be included in this PR?

Reply:
nah, since you haven't touched the function I wouldn't

Reply:
fixed!